Fix spam checker modules documentation example (#9580)
matrix-org/synapse: docs/spam_checker.md

* An instance of `synapse.module_api.ModuleApi`.

It then implements methods which return a boolean to alter behavior in Synapse.

> All the methods are mandatory to be defined.

Reviewer: ```suggestion
All the methods are must be defined.
```

The `ModuleApi` class provides a way for the custom spam checker class to
call back into the homeserver internals.

Additionally, a `parse_config` method is mandatory and receives the plugin config

> dictionary. After possible sanitizing, It must return a dictionary which
> will be passed to `__init__` later.

Reviewer: This doesn't need to return a dictionary, but any object which can go into the `__init__`, I think.

```suggestion
dictionary. After parsing, It must return an object which will be
passed to `__init__` later.
```
